### ISSUE TITLE
Errors adding shipping and checking out if no items from primary shop in cart

### ISSUE THREAD

**Issue by spencern** [reactioncommerce/reaction#2836]
**To Reproduce:**
1. Setup `marketplace` branch with multiple shops using Stripe.
  a. Setup Stripe & Shipping Options for primary shop
  b. Invite and accept 2nd shop
  c. Connect stripe from 2nd shop
2. Create at least one product in 2nd shop.
3. Add the new 2nd shop product you created to a cart in a new session / incognito window. Make sure you don't add any other products from the primary shop to your cart.
4. Attempt to checkout  - make sure to create an account w/ email.

You'll see initially that the shipping methods are not loading. This does not appear to be connected to #2265 / #2738 

If you hack your way through the shipping rates not loading, there is also an error that fires when processing the checkout 
```
 TypeError: Cannot read property '_id' of undefined
      at server/methods/core/cartToOrder.js:90:36`
```
This error is thrown when iterating through the shipping methods if a matching billing method cannot be found. This should probably have a guard, but also reveals a problem in the Shipping array - there is a shipment for the primary shop, but no items or billing records for the primary shop.

**Issues**
There are a number of issues here that we should resolve.
1. We need to make sure we're pruning the shipping records when the cart is updated to ensure that the only shipment records that exist are for shops that are represented by items in the cart.
2. We need to adjust the `cart/setShipmentAddress` method to make sure it's adding the address to ALL shipments and is not creating additional shipping records on the cart where there are no items. **note** there may be other methods that must be updated here as well, it will require some digging.
  The code here https://github.com/reactioncommerce/reaction/blob/marketplace/server/methods/core/cart.js#L714-L740 seems to be the culprit adding a shipment for the primaryShop if none exists.
3. We need to make sure that if the primary shop is not represented in any shipment records, that if `merchantShippingRates` is off, the rates are still loaded from the primary shop. 
  Regarding this particular issue, @zenweasel notes that 
    > so this code is going to be wrong https://github.com/reactioncommerce/reaction/blob/57553e8c18ff328dee9a9ca1ecc101532960bce4/imports/plugins/core/shipping/client/templates/checkout/shipping.js#L37
4. If we get to the `copyCartToOrder` method successfully with more shipping records than billing records, or with records that don't match 1-to-1, we have an error here: https://github.com/reactioncommerce/reaction/blob/marketplace/server/methods/core/cartToOrder.js#L90

These issues appear to have been introduced in #2791 

As they are preventing an order from being placed for any marketplace order that doesn't contain an item from the primaryShop, i'm going ahead and adding the [critical impact] tag.

**Comment by aaronjudd** [comment]
Right now, I can't even checkout with shipping on marketplace. ```Exception while invoking method 'cart/setShipmentMethod' Error: Match error: Match error: Method Name is required
    at exports.check (packages/check.js:56:15)
    at [object Object].cartSetShipmentMethod (server/methods/core/cart.js:549:5)
    at packages/check.js:129:16
    at [object Object]._.extend.withValue (packages/meteor.js:1126:17)
    at Object.exports.Match._failIfArgumentsAreNotAllChecked (packages/check.js:128:41)
    at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1765:18)
    at packages/ddp-server/livedata_server.js:719:19
    at [object Object]._.extend.withValue (packages/meteor.js:1126:17)
    at packages/ddp-server/livedata_server.js:717:46
    at [object Object]._.extend.withValue (packages/meteor.js:1126:17)
    at packages/ddp-server/livedata_server.js:715:46
    at [object Object]._.extend.protocol_handlers.method (packages/ddp-server/livedata_server.js:689:23)
    at packages/ddp-server/livedata_server.js:559:43
Sanitized and reported to the client as: Match failed [400]
```

**Comment by aaronjudd** [comment]
@zenweasel is #2847 the resolution for this critical-impact issue, and it was marked [WIP], and nothing has been updated/changed since then.. is the WIP accurate? can we assign a reviewer?

**Comment by zenweasel** [comment]
@aaronjudd Yes, this is still WIP. I was asked to finish up Stripe Connect first so I am completing that then will return to this.

**Comment by spencern** [comment]
Resolved by #2847
